### DISCUSSION TITLE
HomeAssistant API for clips very slow through reverse proxy?

### DISCUSSION THREAD

**Discussion by Bod1985** [blakeblackshear/frigate#2581] [Ask A Question]
If I load an event clip directly from frigate using the following, it loads very quickly. 

`http://192.168.1.6:5010/api/events/1641138165.128776-3wp18j/clip.mp4`

If I load an event clip from HomeAssistant by local IP like the following, it loads very quickly.

`http://192.168.1.3:8123/api/frigate/notifications/1641138165.128776-3wp18j/clip.mp4`

However, if I load an event clip from HomeAssistant by reverse proxy like the following, it loads extremely slowly - minutes not seconds.

`https://h.domain.uk/api/frigate/notifications/1641138165.128776-3wp18j/clip.mp4`

All other use cases for my reverse proxy are fast and the speed issue is present whether I’m accessing from inside my LAN or over WAN. 

My friend has a similar setup (except Frigate running as an addon inside hass, and not using a NAS for clip storage) and is experiencing the same behaviours.

We are both using Nginx Reverse Proxy. 

Have I/we configured something wrong?

**Comment by anacozero** [comment]
I'm currently having the same issue, except I use Traefik for my reverse proxy. 

Did you ever figure out the cause?

**Comment by freeskier93** [comment]
I know this is old but I think I'm experiencing this issue too using Nabu Casa, and it's incredibly annoying. Clips and recordings play fine everywhere except from Home Assistant media browser or the Lovelace card.

**Reply by Bod1985** [reply]
Im now using Caddy as my reverse proxy and I still can’t play clips and recordings within homeassistant. 

I should have kept the dialog going as I kind of gave up using home assistant card for anything but live view

**Comment by freeskier93** [comment]
> Im now using Caddy as my reverse proxy and I still can’t play clips and recordings within homeassistant.
> 
> I should have kept the dialog going as I kind of gave up using home assistant card for anything but live view

I did a little more troubleshooting and it looks like my issues isn't actually the same. Everything works fine using the regular APIs, even through my domain/Nabu Casa. This for example loads fine:

https://my.public.domain/api/frigate/notifications/1709310861.924537-t7gmq4/clip.mp4

The issue seems specific to Home Assistant Media Browser. Doing an element inspection when playing the clip in Media Browser shows that it uses the following source:

https://my.public.domain/7a64ddac-67ce-4cf2-8fde-0ba87759cf90

Whatever that string is seems to be uniquely generated with every play attempt, and if I attempt to plug that into a browser I just get 404 not found. Now I'm unsure if this is a Frigate issue or Home Assistant issue.

**Reply by Bod1985** [reply]
Thanks for doing that research.

I’ll compare my situation when I get some time as it could be that my triage the other year was throwing up red herrings and we are in the same boat.
